feg/gateway/services/session_proxy/credit_control/gy/gy_client_test.go

```diff
@@ -398,19 +448,22 @@ func getGyGlobalConfig(apnFilter, apnOverwrite string) *gy.GyGlobalConfig {
 	}
 }
 
-func startServer(client *diameter.DiameterClientConfig, server *diameter.DiameterServerConfig, initMethod gy.InitMethod) *mock_ocs.OCSDiamServer {
+func startServer(client *diameter.DiameterClientConfig, server *diameter.DiameterServerConfig, initMethod gy.InitMethod, action gy.FinalUnitAction) *mock_ocs.OCSDiamServer {
 	serverStarted := make(chan struct{})
 	var ocs *mock_ocs.OCSDiamServer
 	go func() {
 		log.Printf("Starting server")
 		ocs = mock_ocs.NewOCSDiamServer(
 			client,
 			&mock_ocs.OCSConfig{
-				MaxUsageOctets: &fegprotos.Octets{TotalOctets: returnedOctets},
-				MaxUsageTime:   1000,
-				ValidityTime:   validityTime,
-				ServerConfig:   server,
-				GyInitMethod:   initMethod,
+				MaxUsageOctets:  &fegprotos.Octets{TotalOctets: returnedOctets},
+				MaxUsageTime:    1000,
+				ValidityTime:    validityTime,
+				ServerConfig:    server,
+				GyInitMethod:    initMethod,
+				RestrictRules:   []string{restrictRule},
```

Comment:
what about changing `OCSConfig` to include a `FinalUnitAction` instead of having unstructure config type?


(note i didn't use the right types)
```
type OCSConfig struct {
	MaxUsageOctets  	 *protos.Octets
	MaxUsageTime    	 uint32
	ValidityTime    	 uint32
	ServerConfig    	 *diameter.DiameterServerConfig
	GyInitMethod    	 gy.InitMethod
	UseMockDriver   	 bool
	FinaalUnitIndication FinalUnitIndication
}


struct FinalUnitInidcation {
	finalUnitAction protos.FinalUnitAction,
	redirectAddr	string:,
	restrict	 	[]string, 

}
```